modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/java/com/liferay/dynamic/data/mapping/web/internal/item/selector/DDMFormInstanceItemSelectorView.java

```diff
@@ -0,0 +1,345 @@
+import com.liferay.item.selector.criteria.info.item.criterion.InfoItemItemSelectorCriterion;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.dao.search.SearchContainer;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.portlet.SearchOrderByUtil;
+import com.liferay.portal.kernel.security.permission.ResourceActionsUtil;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.HtmlUtil;
+import com.liferay.portal.kernel.util.JavaConstants;
+import com.liferay.portal.kernel.util.LocalizationUtil;
+import com.liferay.portal.kernel.util.OrderByComparator;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.io.IOException;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+import javax.portlet.PortletRequest;
+import javax.portlet.PortletURL;
+
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Mikel Lorza
+ */
+@Component(
+	property = "item.selector.view.order:Integer=200",
+	service = ItemSelectorView.class
+)
+public class DDMFormInstanceItemSelectorView
+	implements InfoItemSelectorView,
+			   ItemSelectorView<InfoItemItemSelectorCriterion> {
+
+	@Override
+	public String getClassName() {
+		return DDMFormInstance.class.getName();
+	}
+
+	@Override
+	public Class<InfoItemItemSelectorCriterion>
+		getItemSelectorCriterionClass() {
+
+		return InfoItemItemSelectorCriterion.class;
+	}
+
+	@Override
+	public List<ItemSelectorReturnType> getSupportedItemSelectorReturnTypes() {
+		return _supportedItemSelectorReturnTypes;
+	}
+
+	@Override
+	public String getTitle(Locale locale) {
+		return _language.get(locale, "forms");
+	}
+
+	@Override
+	public void renderHTML(
+			ServletRequest servletRequest, ServletResponse servletResponse,
+			InfoItemItemSelectorCriterion infoItemItemSelectorCriterion,
+			PortletURL portletURL, String itemSelectedEventName, boolean search)
+		throws IOException, ServletException {
+
+		_itemSelectorViewDescriptorRenderer.renderHTML(
+			servletRequest, servletResponse, infoItemItemSelectorCriterion,
+			portletURL, itemSelectedEventName, search,
+			new DDMFormInstanceSelectorViewDescriptor(
+				(HttpServletRequest)servletRequest,
+				infoItemItemSelectorCriterion, portletURL));
+	}
+
+	private static final List<ItemSelectorReturnType>
+		_supportedItemSelectorReturnTypes = Collections.singletonList(
+			new InfoItemItemSelectorReturnType());
```

Comment:
As I can figure out, the ItemSelectorReturnType is used to filter the ItemSelectorView's  in the BaseItemSelectorCriterionHandler class. Also as I am using the default item selector renderer implementation to render the html of the item selector, and there in the view_item_selector_view_descriptor.jsp we are using the ItemSelectorReturnType in the line 227 of the file and I am in conversation with clara to know how it affects to the frontend. Maybe we have to add the UUIDItemSelectorReturnType to avoid a better return type but I have to confirm with the frontend people.

Reply:
The Lima people are the ones that maintain this framework (no the frontend people). But we are the ones that just read the UUIDItemSelectorReturnType, in order to support the InfoItemItemSelectorReturnType, @claraizquierdo and @Stephy6 need to change the code of our Segment Editor Frontend.

Reply:
The ItemSelectorReturnType is an Interface.

The InfoItemItemSelectorReturnType is the implementation you are using, but you should use UUIDItemSelectorReturnType.

Reply:
Hi @claraizquierdo , 

When you have time I would like to know if this implementation fit for frontend site when the user select an item. So, the better way is to test it with you to know how is working the returnType that is already in the jsp view_item_selector_view_descriptor.jsp (line 227)  .

Thanks

Reply:
If I have to use it , I can use it, but I would like to know how the frontend is working with both return type's because I don' know it.